### PR TITLE
`unnecessary_placeholder_linter()` covers a few other ops

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Part of #1657

### CODE REVIEW

File: tests/testthat/test-unnecessary_placeholder_linter.R [hunk 1]
```diff
@@ -1,25 +1,40 @@
...
-  # . used in position other than first --> ok
-  expect_lint("x %>% foo(y, .)", NULL, unnecessary_placeholder_linter())
-  # ditto for nested usage
-  expect_lint("x %>% foo(y, bar(.))", NULL, unnecessary_placeholder_linter())
-  # . used twice --> ok
-  expect_lint("x %>% foo(., .)", NULL, unnecessary_placeholder_linter())
-  # . used as a keyword argument --> ok
-  expect_lint("x %>% foo(arg = .)", NULL, unnecessary_placeholder_linter())
-})
+skip_if_not_installed("patrick")
```
Comment (MichaelChirico) [inline tests/testthat/test-unnecessary_placeholder_linter.R:1]: hmm at this point I think we have enough of our test suite using Patrick that we should make it required & skip this overhead...

Comment (MichaelChirico) [reply tests/testthat/test-unnecessary_placeholder_linter.R:1]: #1682

File: R/unnecessary_placeholder_linter.R [hunk 2]
```diff
@@ -33,7 +33,7 @@
 unnecessary_placeholder_linter <- function() {
   # TODO(michaelchirico): handle R4.2.0 native placeholder _ as well
   xpath <- "
-  //SPECIAL[text() = '%>%']
+  //SPECIAL[text() = '%>%' or text() = '%T>%' or text() = '%<>%']
```
Comment (MichaelChirico) [inline R/unnecessary_placeholder_linter.R:36]: Note which are covered in the documentation?

Comment (IndrajeetPatil) [reply R/unnecessary_placeholder_linter.R:38]: Good point. Done.

We can revise this once the native pipe is also supported.

**Review by MichaelChirico — APPROVED** [review]

**Pull request merged**